apps/analysis/pipelines.py

```diff
@@ -0,0 +1,80 @@
+from apps.analysis.steps.loaders import ResourceTextLoader
+from apps.analysis.steps.parsers import WhatsappParser
+
+from .core import ParamsForm, Pipeline
+from .steps.filters import TimeseriesFilter
+from .steps.processors import AssistantStep, LlmCompletionStep
+from .steps.splitters import TimeseriesSplitter
+
+LLM_PIPE = "llm"
+
+TEXT_DATA_PIPE = "text_data"
+
+SOURCE_PIPELINES = {
+    TEXT_DATA_PIPE: Pipeline(
+        [
+            ResourceTextLoader(),
+        ],
+        "Text Data",
+        "Load text data from a file",
+    ),
+    "filtered_whatsapp_data": Pipeline(
+        [
+            ResourceTextLoader(),
+            WhatsappParser(),
+            TimeseriesFilter(),
+        ],
+        "Filtered WhatsApp Data",
+        "Load WhatsApp data from a file and filter it by date.",
+    ),
+    "split_whatsapp_data": Pipeline(
+        [
+            ResourceTextLoader(),
+            WhatsappParser(),
+            TimeseriesSplitter(),
+        ],
+        "Split WhatsApp Data",
+        "Load whatsApp data from a file and split it into chunks by date.",
+    ),
+}
+
+
+PIPELINES = {
+    LLM_PIPE: Pipeline(
```

Comment:
> nit: the other keys in this dict aren't constants like this one

Same here. Was there a reason for this?

Reply:
These were the ones that I accessed in tests so I made them contstants. I'm going to refactor this code later anyway.